components/usage/pkg/db/cost_center.go

```diff
@@ -15,13 +15,20 @@ import (
 
 var CostCenterNotFound = errors.New("CostCenter not found")
 
+type BillingStrategy string
+
+const (
+	CostCenter_Stripe BillingStrategy = "stripe"
+	CostCenter_Other  BillingStrategy = "other"
+)
+
 type CostCenter struct {
-	ID            AttributionID `gorm:"primary_key;column:id;type:char;size:36;" json:"id"`
-	SpendingLimit int32         `gorm:"column:spendingLimit;type:int;default:0;" json:"spendingLimit"`
-	LastModified  time.Time     `gorm:"->:column:_lastModified;type:timestamp;default:CURRENT_TIMESTAMP(6);" json:"_lastModified"`
+	ID              AttributionID   `gorm:"primary_key;column:id;type:char;size:36;" json:"id"`
+	CreationTime    VarcharTime     `gorm:"primary_key;column:creationTime;type:varchar;size:255;" json:"creationTime"`
+	SpendingLimit   int32           `gorm:"column:spendingLimit;type:int;default:0;" json:"spendingLimit"`
+	BillingStrategy BillingStrategy `gorm:"column:billingStrategy;type:varchar;size:255;" json:"billingStrategy"`
 
-	// deleted is restricted for use by db-sync
-	_ bool `gorm:"column:deleted;type:tinyint;default:0;" json:"deleted"`
+	LastModified time.Time `gorm:"->:column:_lastModified;type:timestamp;default:CURRENT_TIMESTAMP(6);" json:"_lastModified"`
```

Comment:
out of curiosity: are the gorm definitions manually crafted, or is there a tool to sync them?

Reply:
manual

Reply:
There's a tool which can generate them from the DB schema. I used those initially to set up the definitions but often requires some manual tweaking.

Reply:
Something I actually forgot to mention in yesterday's retro is that I'm getting a little tired of maintaining/writing all the different technical representations (DB & protobuf for TS&Go) of the same shapes. Feels like a lot of code (e.g. this PR) is really just about shoveling data through the stack by copying it into different formats. 😓

Reply:
That's fair. Codegen can help but it's also important to keep the API shape independent of the shape of the DB. It tends to start very similar, but over time changes. The DB is expected to evolve while the API is expected to retain compatibility.

Reply:
Yes, I think it would already help if we didn't need to duplicate API to route dashboard requests through server. Also, I think the manual `sugar.ts` code could maybe be generated.